api/files/views.py

```diff
@@ -171,3 +178,41 @@ def get_serializer_context(self):
         context = JSONAPIBaseView.get_serializer_context(self)
         context['file'] = self.file
         return context
+
+
+class FileCedarMetadataRecordsList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin, FileCedarMetadataRecordMixin):
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        PermissionWithGetter(ContributorOrPublic, 'target'),
+    )
+    required_read_scopes = [CoreScopes.CEDAR_METADATA_RECORD_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = CedarMetadataRecordsListSerializer
+
+    view_category = 'files'
+    view_name = 'file-cedar-metadata-records-list'
+
+    def get_default_queryset(self):
+        self.get_file()
+        file_records = None
+        file_id_or_guid = self.kwargs['file_id_or_guid']
+        try:
+            Guid.objects.get(_id=file_id_or_guid)
+            file_records = CedarMetadataRecord.objects.filter(guid___id=file_id_or_guid)
+        except Guid.DoesNotExist:
+            file = BaseFileNode.load(file_id_or_guid)
+            if file:
+                guid = file.get_guid()
+                if guid:
+                    file_records = CedarMetadataRecord.objects.filter(guid___id=guid._id)
+        if not file_records:
+            return CedarMetadataRecord.objects.none()
```

Comment:
This, coupled with `get_file_guid_or_id` defined in `./serializers.py`, seems much more complicated than necessary. A file can only have CMRs if it has a Guid, so this could be simplified with something like
```
obj = self.get_file()
guid = obj.get_guid()
if not guid:
    return CedarMetadataRecord.objects.none()
user_auth = utils.get_user_auth(self.request)
return CedarMetadataRecord.objects.filter(guid___id=guid._id).can_view(auth=user_auth)  # Aspirational can_view
```
This obviates the `get_file_guid_or_id` helper in `./serializers` and normalizes the url structure

Reply:
~~I think during the implementation, it was requested that we should support the case when a `file_id`.~~

Ah, I see, we were trying to fix a permission issue so I added the `self.get_file()` as the fix but didn't touch the rest. Let me try to leverage what we already have from `self.get_file()`.